### PR TITLE
ENH: Add `max_loading` on lines and transformers

### PR DESCRIPTION
**Pull request by benoit9126**
Implementation of #279

### CODE REVIEW

File: roseau/load_flow/models/lines/lines.py [hunk 1]
```diff
@@ -303,16 +332,28 @@ def res_power_losses(self) -> Q_[ComplexArray]:
...
 
+    @property
+    def res_loading(self) -> Q_[FloatArray] | None:
+        """Get the loading of the line (unitless)."""
+        amp = self._parameters._ampacities
+        if amp is None:
+            return None
+        currents1, currents2 = self._res_currents_getter(warning=True)
+        i_max = amp * self._max_loading
+        return Q_(np.maximum(abs(currents1), abs(currents2)) / i_max, "")
```
Comment (Saelyos) [inline roseau/load_flow/models/lines/lines.py:343]: Maybe `@ureg_wraps` can be used for this property ?

File: roseau/load_flow/models/transformers/transformers.py [hunk 2]
```diff
@@ -291,16 +320,27 @@ def res_power_losses(self) -> Q_[complex]:
...
+    @property
+    def res_loading(self) -> Q_[float] | None:
+        """Get the loading of the transformer (unitless)."""
+        sn = self._parameters._sn
+        if sn is None:
+            return None
+        powers1, powers2 = self._res_powers_getter(warning=True)
+        s_max = sn * self._max_loading
+        return Q_(max(abs(powers1.sum()), abs(powers2.sum())) / s_max, "")
+
```
Comment (Saelyos) [inline roseau/load_flow/models/transformers/transformers.py:332]: Same

File: roseau/load_flow/io/dict.py [hunk 3]
```diff
@@ -649,44 +649,60 @@ def v2_to_v3_converter(data: JsonDict) -> JsonDict:
...
     old_transformers_params = data.get("transformers_params", [])
     transformers_params = []
     for transformer_param_data in old_transformers_params:
-        if up := transformer_param_data.pop("uhv", None) is not None:
+        if (up := transformer_param_data.pop("uhv", None)) is not None:
             transformer_param_data["up"] = up
-        if us := transformer_param_data.pop("ulv", None) is not None:
+        if (us := transformer_param_data.pop("ulv", None)) is not None:
             transformer_param_data["us"] = us
+        transformer_param_data.pop("max_power", None)
```
Comment (alihamdan) [inline roseau/load_flow/io/dict.py:660]: The value of `max_power` (when not None) should be used to compute the `max_loading` value on transformers that use this Parameters object. Example:
```python
old_tp = {"id": "old_tp1", "sn": 100_000, "max_power": 80_000, ...}
old_tr = {"id": "old_tr1", "params_id":  "old_tp1", ...}

new_tp = {"id": "old_tp1", "sn": 100_000, ...}  # no more max_power
new_tr = {"id": "old_tr1", "params_id":  "old_tp1", "max_loading": old_tp["max_power"]/old_tp["sn"], ...}

```

File: roseau/load_flow/io/dict.py [hunk 4]
```diff
@@ -649,44 +649,60 @@ def v2_to_v3_converter(data: JsonDict) -> JsonDict:
...
+    lines = []
+    for line_data in old_lines:
+        line_data["max_loading"] = 1
+        lines.append(line_data)
+
+    # Add max_loading to transformers
+    old_transformers = data.get("transformers", [])
+    transformers = []
+    for transformer_data in old_transformers:
+        transformer_data["max_loading"] = 1
```
Comment (alihamdan) [inline roseau/load_flow/io/dict.py:694]: See above, this value is not always `1` if `max_power` was defined

File: roseau/load_flow/io/tests/test_dict.py [hunk 5]
```diff
@@ -166,13 +166,11 @@ def test_to_dict():
 
     # Dict content
     transformer2.parameters = tp1
-    tp1.max_power = 180_000
```
Comment (alihamdan) [inline roseau/load_flow/io/tests/test_dict.py:169]: Maybe test `transformer2.max_loading` here instead?

File: roseau/load_flow/models/lines/lines.py [hunk 6]
```diff
@@ -221,11 +227,34 @@ def y_shunt(self) -> Q_[ComplexArray]:
...
+        """The ampacities of the line (in A)."""
+        # Do not add a setter. The user must know that if they change the ampacities, it changes
         # for all lines that share the parameters. It is better to set it on the parameters.
-        return self._parameters.max_currents
+        return self._parameters.ampacities
+
+    @property
+    def max_currents(self) -> Q_[FloatArray] | None:
+        """The maximum current of the line (in A). It takes into account the `max_loading` of the line and the
+        ampacities` of the parameters."""
```
Comment (alihamdan) [inline roseau/load_flow/models/lines/lines.py:254]:
```suggestion
        `ampacities` of the parameters."""
```

File: roseau/load_flow/models/lines/lines.py [hunk 7]
```diff
@@ -72,6 +73,10 @@ def __init__(
 
             geometry:
                 The geometry of the line i.e. the linestring.
+
+            max_loading:
```
Comment (alihamdan) [inline roseau/load_flow/models/lines/lines.py:77]: Say that it is not used in the load flow (as we do for all other optional parameters)

File: roseau/load_flow/models/transformers/parameters.py [hunk 8]
```diff
@@ -353,7 +339,6 @@ def from_power_factory(
         pc: float | Q_[float],
```
Comment (alihamdan) [inline roseau/load_flow/models/transformers/parameters.py:339]: I didn't notice this in your last PR but the parameters `uhv` and `ulv` for the `from_power_factory` method must not be renamed to `up` and `us`. The whole point of this method is that it uses the power factory notation and parameters and converts to them to the proper RLF object.

**Review by alihamdan — CHANGES_REQUESTED** [review]
Thank you. Noticed a few thinks.

**Review by alihamdan — APPROVED** [review]

**Pull request merged**